### ISSUE TITLE
[TW-1828] don't limit lines/rows when piped

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1844]
_Alois Mahdal on 2016-07-07T18:56:23Z says:_ 

 With lot of pending tasks with long descriptions (eg. URLs), command like:

{code}
task > foo
{code}

will apply line and column limit, even if the output is not TTY, ie. not a terminal.

(at)pbeckingham's explanation from IRC:

bq. [...] If the output is piped/redirected, there is no longer any 'width', so it falls back to defaults. [...]

makes sense, but question is, since there's no such thing as width/height limitations for files/pipes, why is such behavior even triggered?

In other words, it would make much more sense if command above did not apply any limit at all (unless explicitly specified); avoiding need to fiddle with `limit` or `rc.detection` or `rc.defaultwidth`.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2016-07-07T18:56:23Z
Modified: 2017-01-16T20:34:38Z
```

**Comment by taskwarrior** [comment]
_Alois Mahdal on 2016-07-07T19:01:09Z says:_ 

 Case in point: to make easy use of ansi2html utility, I have to use this syntax:

{code}
task rc._forcecolor:yes limit:0 burndown | ansi2html > burndown.html
{code}

otherwise (without `limit:0`), TW trims the list to seemingly arbitrary number of lines.

With this feature, the `limit:0` could be dropped.  (Similar case could be made with regards to width limitations.)

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2017-01-16T20:34:38Z says:_ 

 Closing.  The workarounds are perfectly reasonable, and the current implementation was deliberately done this way.  There are multiple good values for piped output, and choosing one over another is somewhat arbitrary, and has already been done.
